src/utils/util.c

```diff
@@ -72,3 +86,88 @@ void check_if_running()
                 exit(EXIT_FAILURE);
         }
 }
+
+void check_argument_length(char *arg, int type)
+{ 
+        size_t len = strlen(arg) + 1;
+        if (type == TRUSTEDLIST) {
+                dprintf("the length of trusted list is %ld\n", len);
+                if (len > MAX_TRUSTED_LIST_LENGTH) {
+                        fprintf(stderr, "The trusted list characters exceeds the limit of %d\n", MAX_TRUSTED_LIST_LENGTH);
+                        exit(EXIT_FAILURE);
+                }
+        }
+
+        if (type == CANARYTOKEN) {
+                dprintf("the length of canary tokeb is %ld\n", len);
+                if (len > MAX_CANARY_TOKEN_LENGTH) {
+                        fprintf(stderr, "The canary token characters exceeds the limit of %d\n", MAX_CANARY_TOKEN_LENGTH);
+                        exit(EXIT_FAILURE);
+                }
+        }
+}
+
+void config_file_handler(char *cnrytkn, char *trstdlst)
+{
+        FILE* fp;
+        char errbuf[200];
+        
+        char config_file[PATH_MAX];
+
+#ifdef TESTS
+        char cwd[PATH_MAX];
+        if (getcwd(cwd, sizeof(cwd)) != NULL) {
+                dprintf("Current working dir for testing: %s\n", cwd);
+                sprintf(config_file, "%s/%s", cwd, test_config_file);
+        } else {
+                fprintf(stderr, "TESTING ERROR: not possible to retrieve current directory\n");
+                exit(EXIT_FAILURE);
+        }
+#else
+        sprintf(config_file, "%s/%s", getenv("HOME"), CONFIG_FILE);
+#endif
+
+        fp = fopen(config_file, "r");
+        if (!fp) {
+                dprintf("No config file at %s\n", config_file);
+                show_help();
+        }
+
+        toml_table_t* conf = toml_parse_file(fp, errbuf, sizeof(errbuf));
```

Comment:
When declaring pointer data or a function that returns a pointer type, the preferred use of * is adjacent to the data name or function name and not adjacent to the type name